### PR TITLE
Contributing Guide Updates

### LINKED ISSUE #29758: [BUG] Documentation Inconsistency in contributing.md
### Library name and version

N/A

### Describe the bug

The documentation in contributing.md indicates that the input files should be of the form `https://github.com/Azure/azure-rest-api-specs/blob/main/specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/stable/2022-05-13/confidentialledger.json` when I believe it should be of the form `https://github.com/Azure/azure-rest-api-specs/blob/61001b68a8aa743f0dd890224c6b5cb130ef006e/specification/confidentialledger/resource-manager/confidentialledger.json`. Thank you!

### Expected behavior

N/A

### Actual behavior

N/A

### Reproduction Steps

N/A

### Environment

_No response_

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to make minor formatting and content updates to the contributing guide.  Included changes:

- Fixes for some awkward phrasing
- Expanding some abbreviations for readability
- Updates to the test matrix to reflect the current state
- Update "master" to "main" for branch references
- Make the pattern of using a stable hash for codegen explicit
- Minor typo fixes

# References and Resources

- [[BUG] Documentation Inconsistency in contributing.md (#29758)](https://github.com/Azure/azure-sdk-for-net/issues/29758)

### CODE REVIEW

**Review by heaths — APPROVED** [review]
A couple instances for a small change, but otherwise LGTM. Did you also check there were no bookmark links to section headers you changed in this markdown file? I don't recall any, but from context here can't tell easily.

**Review by ArthurMa1978 — APPROVED** [review]

**Comment by jsquire** [comment]
> Did you also check there were no bookmark links to section headers you changed in this markdown file? I don't recall any, but from context here can't tell easily.

Thank you!   I forgot to check and there was one impacted.   Fixed.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -462,7 +462,7 @@ See [Data Plane Quick Start Tutorial](https://github.com/Azure/azure-sdk-for-net
 Before a pull request will be considered by the Azure SDK team, the following requirements must be met:
 
 - Prior to issuing the pull request:
-  - All code must have completed any necessary legal signoff for being publicly viewable (Patent review, JSR review, etc.)
+  - All code must have completed any necessary legal sign off for being publicly viewable (Patent review, JSR review, etc.)
```
[the remark this answers is not in the document]
Comment (jsquire) [reply CONTRIBUTING.md]: Thanks for including the reasoning; I learned me something today.

[the remark this answers is not in the document]
Comment (heaths) [reply CONTRIBUTING.md]: Given my history, I see this often with "setup" (noun) vs. "set up" (verb). `/sign\s?off/` is similar. In most cases it's not an issue, but I figured may as well be right in more customer-facing documentation.

**Pull request merged**